File: app/src/main/java/com/pr0gramm/app/ui/views/BusyIndicator.java
package com.pr0gramm.app.ui.views;

import android.content.Context;
import android.support.v4.content.ContextCompat;
import android.util.AttributeSet;

import com.pnikosis.materialishprogress.ProgressWheel;

import static com.pr0gramm.app.services.ThemeHelper.primaryColor;

/**
 */
public class BusyIndicator extends ProgressWheel {
    public BusyIndicator(Context context) {
        super(context);
        init();
    }

    public BusyIndicator(Context context, AttributeSet attrs) {
        super(context, attrs);
        init();
    }

    private void init() {
        setBarColor(ContextCompat.getColor(getContext(), primaryColor(getContext())));
        spin();
    }
}

File: app/src/main/java/com/pr0gramm/app/ui/views/BusyIndicator.kt
package com.pr0gramm.app.ui.views

import android.content.Context
import android.util.AttributeSet
import com.pnikosis.materialishprogress.ProgressWheel
import com.pr0gramm.app.services.ThemeHelper.accentColor
import com.pr0gramm.app.util.getColorCompat

/**
 * A busy indicator in the apps accent color.
 */
class BusyIndicator @JvmOverloads constructor(context: Context, attrs: AttributeSet? = null) :
        ProgressWheel(context, attrs) {

    init {
        barColor = context.getColorCompat(accentColor)
        spin()
    }
}
